llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -152,28 +161,31 @@ bool AIELegalizerHelper::legalizeG_BUILD_VECTOR(LegalizerHelper &Helper,
   MIRBuilder.buildUndef(Src);
 
   const AIEBaseInstrInfo *II = ST.getInstrInfo();
-  MachineOperand *OperandBegin = MI.operands_begin(),
-                 *Operand = MI.operands_end() - 1;
-  while (Operand != OperandBegin) {
-    Register Reg = Operand->getReg();
+  MachineOperand *OperandBegin = MI.operands_begin() + 1,
+                 *Operand = MI.operands_begin() + 1,
+                 *OperandEnd = MI.operands_end();
+  while (Operand != OperandEnd) {
```

Comment:
If there's any way to write this as a range-based for, I would be happier. Aren't we traversing all uses of MI here?